File: 백준/Silver/9012. 괄호/괄호.cc
#include <iostream>
#include <string>

using namespace std;

int main() {
	int N;
	cin >> N;

	while (N--) {
		string s;
		cin >> s;
		int cnt = 0;
		bool isValid = true;

		for (char c : s) {
			if (c == '(') {
				cnt++;
			}
			else if (c == ')') {
				cnt--;
				if (cnt < 0) { 
					isValid = false;
					break;
				}
			}
		}

		if (isValid && cnt == 0) cout << "YES\n";
		else cout << "NO\n";
	}

	return 0;
}

File: 백준/Silver/9012. 괄호/괄호.py
n = int(input())

for _ in range(n):
    s = input()
    cnt = 0
    is_valid = True

    for c in s:
        if c == '(':
            cnt += 1
        elif c == ')':
            cnt -= 1
            if cnt < 0:
                is_valid = False
                break

    if is_valid and cnt == 0:
        print("YES")
    else:
        print("NO")
